server/boutique-hotel-technikum/src/main/java/com/example/boutiquehoteltechnikum/services/RoomService.java

```diff
@@ -3,31 +3,41 @@
 import com.example.boutiquehoteltechnikum.models.RoomEntity;
 import com.example.boutiquehoteltechnikum.repositories.RoomRepository;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpStatus;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
-import org.springframework.web.server.ResponseStatusException;
 
+import java.sql.Date;
 import java.util.List;
 
 @Service
 public class RoomService {
 
     @Autowired private RoomRepository roomRepository;
 
-    public List<RoomEntity> getRooms() {
-        return roomRepository.findAll();
+    public Page<RoomEntity> getRooms(int limit, int offset) {
+        Pageable pageable = PageRequest.of(offset / limit, limit);
+        return roomRepository.findAll(pageable);
     }
-
-    public RoomEntity getRoom(int id) {
-        if (roomRepository.findById(id).isPresent()) {
-            return roomRepository.findById(id).get();
-        } else {
-            throw new ResponseStatusException(HttpStatus.NOT_FOUND, String.format("Room with id %d not found", id));
+    public List<RoomEntity> getAvailableRooms(String startDate, String endDate) {
+        Date start = Date.valueOf(startDate);
+        Date end = Date.valueOf(endDate);
+        return roomRepository.findAvailableRooms(start, end);
+    }
```

Comment:
Please combine these two like `getRoomById` and if no start or end date is specified, all Rooms are loaded but if many rooms are available, they are returned with limit and offset (Important: total is then only the number of available rooms in the specified time range)

Reply:
Done + rewrote the tests.